File: src/main/java/org/pl/parser/ast/BinaryOperationNode.java
package org.pl.parser.ast;

public class BinaryOperationNode implements INode {
    public final INode leftNode;
    public final INode rightNode;
    public final BinaryOperation op;

    public BinaryOperationNode(INode leftNode, INode rightNode, BinaryOperation op) {
        this.leftNode = leftNode;
        this.rightNode = rightNode;
        this.op = op;
    }

    @Override
    public String toString() {
        return "BinaryOperationNode{" +
                "leftNode=" + leftNode +
                ", rightNode=" + rightNode +
                ", op=" + op +
                '}';
    }
}

File: src/main/kotlin/org/leaflang/parser/ast/BinaryOperationNode.kt
package org.leaflang.parser.ast

import org.leaflang.parser.utils.NodePosition

/**
 * Node with two child nodes and a binary operation.
 */
class BinaryOperationNode(override val position: NodePosition,
                          val leftNode: INode,
                          val rightNode: INode,
                          val op: BinaryOperation) : INode {
    override fun toString() = "BinaryOperationNode{leftNode=$leftNode, rightNode=$rightNode, op=$op}"
}